eng/tools/ToolDescriptionEvaluator/Program.cs

```diff
@@ -16,110 +15,169 @@ class Program
                 }
+            }
+            else if (args[i] == "--server" && i + 1 < args.Length)
+            {
+                serverName = args[i + 1];
 
-                if (string.IsNullOrEmpty(toolDescription) || prompts.Count == 0)
+                if (!ValidServers.ContainsKey(serverName))
                 {
-                    Console.WriteLine("❌ Error: --validate mode requires exactly one --tool-description and at least one --prompt argument");
-                    Console.WriteLine("Examples:");
-                    Console.WriteLine("  # Single prompt validation:");
-                    Console.WriteLine("  --validate --tool-description \"Lists all storage accounts\" --prompt \"show me my storage accounts\"");
-                    Console.WriteLine("  # Multiple prompt validation:");
-                    Console.WriteLine("  --validate --tool-description \"Lists all storage accounts\" --prompt \"show me my storage accounts\" --prompt \"list my storage accounts\" --prompt \"what storage accounts do I have\"");
-                    Environment.Exit(1);
+                    throw new ArgumentException($"Invalid server name: {serverName}. Allowed values are 'Azure' or 'Fabric'.");
                 }
+            } else if (args[i] == "--server-exe" && i + 1 < args.Length)
+            {
+                serverExePath = args[i + 1];
 
-                await RunValidationModeAsync(toolDir, toolDescription, prompts, isCiMode);
+                if (!File.Exists(serverExePath))
+                {
+                    throw new FileNotFoundException($"The specified server executable path does not exist: {serverExePath}");
+                }
+            }
+        }
 
-                return;
+        try
+        {
+            if (testSingleToolMode)
+            {
+                WarnArgumentIsIgnored(areaFilter, "--area", "--test-single-tool");
+                WarnArgumentIsIgnored(serverName, "--server", "--test-single-tool");
+                WarnArgumentIsIgnored(serverExePath, "--server-exe", "--test-single-tool");
+                WarnArgumentIsIgnored(customPromptsFile, "--prompts-file", "--test-single-tool");
+                WarnArgumentIsIgnored(customOutputFileName, "--output-file-name", "--test-single-tool");
+
+                if (string.IsNullOrEmpty(testToolDescription))
+                {
+                    throw new ArgumentException("--test-single-tool mode requires exactly one --tool-description argument.");
+                }
+
+                if (testPrompts.Count == 0)
+                {
+                    throw new ArgumentException("--test-single-tool mode requires at least one --prompt argument.");
+                }
+            }
+            else if (!string.IsNullOrEmpty(testToolDescription) || testPrompts.Count > 0)
+            {
+                throw new ArgumentException("--tool-description and --prompt arguments are only valid in --test-single-tool mode.");
```

Comment:
Seems odd that some argument combinations are ignored while others result in errors. Should all the warnings about ignored arguments above be replaced with errors instead?

Reply:
Not sure what's the sweet spot with this kind of tools, so I'm open to changing my mind if you feel it's better to fail fast.